File: abc083-b/java/Main.java
import java.util.Scanner;
import java.util.stream.IntStream;

public class Main {
    public static int getDigitSum(int n) {
        int sum = 0;
        while (n > 0) {
            sum += n % 10;
            n /= 10;
        }
        return sum;
    }

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            int rangeMax = scanner.nextInt();
            int digitSumMin = scanner.nextInt();
            int digitSumMax = scanner.nextInt();

            int sum = IntStream.rangeClosed(1, rangeMax)
                    .filter(n -> digitSumMin <= getDigitSum(n) && getDigitSum(n) <= digitSumMax)
                    .sum();

            System.out.println(sum);
        }
    }
}

File: abc083-b/rust/main.rs
#!/usr/bin/env rust-script

//! ```cargo
//! [dependencies]
//! proconio = "0.4.5"
//! ```

use proconio::input;

fn main() {
    input! {
        range_max: i32,
        digit_sum_min: i32,
        digit_sum_max: i32
    }

    let sum = (1..=range_max)
        .filter(|&x| {
            let sum = x.to_string().chars().map(|c| c.to_digit(10).unwrap()).sum::<u32>() as i32;
            digit_sum_min <= sum && sum <= digit_sum_max
        })
        .sum::<i32>();

    println!("{}", sum);
}
